showcase/app/controllers/page-components/flyout.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Controller from '@ember/controller';
+import { action } from '@ember/object';
+import { deepTracked } from 'ember-deep-tracked';
+
+import type { PageComponentsFlyoutModel } from 'showcase/routes/page-components/flyout';
+
+export default class PageComponentsFlyoutController extends Controller {
+  declare model: PageComponentsFlyoutModel;
+
+  @deepTracked flyouts = {
+    mediumFlyoutActive: false,
+    largeFlyoutActive: false,
+    dropdownInitiatedFlyoutActive: false,
+    dropdownInitiatedWithReturnedFocusFlyoutActive: false,
+    deactivateFlyoutOnCloseActive: false,
+    deactivateFlyoutOnDestroyActive: false,
+    deactivateFlyoutOnSubmitActive: false,
+    deactivateFlyoutOnSubmitValidationError: false,
```

Comment:
[todo] sorry I missed this one too: it should be separate from the list of "active" states